src/main/java/vivatech/init/VivatechBlocks.java

```diff
@@ -12,7 +20,7 @@
     public static final EnergyConduitBlock ENERGY_CONDUIT;
     public static final CoalGeneratorBlock COAL_GENERATOR;
     public static final CrusherBlock CRUSHER;
-    public static final ElectricFurnaceBlock ELECTRIC_FURNACE;
+    public static final AbstractTieredMachineBlock[] ELECTRIC_FURNACE;
```

Comment:
Oh. Try to cast

Reply:
It's very possible I'm doing something wrong, but all my attempts to cast the array result in this error ` java.lang.ClassCastException: [Lvivatech.block.AbstractTieredMachineBlock; cannot be cast to [Lvivatech.block.ElectricFurnaceBlock;` despite the type in the array being an instance of electric furnace. 
I suspect it's something to do with casting arrays but I haven't been able to figure out what's wrong.

Reply:
Cast while generating an array using generics. Arrays themselves cannot be really casted iirc

Reply:
Can't you just use an immutable list (like Guava's `ImmutableList`) there?
Or with arrays, you could make `TierHelper.fillTieredBlockArray` generic.

Reply:
Yep. That’s what I’m asking for

Reply:
Arrays absolutely can be cast, but they're type-checked more strictly. The problem here is that it's a widening cast (putting an `ElectricFurnaceBlock[]` into an `AbstractTieredMachineBlock[]` field) Look at this another way: What if you were to put a non-furnace block into that array now that it has wider scope? If it wasn't an array, that danger wouldn't be there and you'd be fine to do it. On the other hand, if you were to put an `Object[]` there, anything you could stuff into an `AbstractTieredMachineBlock[]` can be safely stuffed into an `Object[]`, so that conversion would go through fine.

Welcome to arrays being more reified and better type-checked than any other java object

Reply:
But EFB extends ATMB. So it shall be a narrowing cast